app/src/main/kotlin/at/bitfire/davdroid/sync/Syncer.kt

```diff
@@ -133,26 +138,30 @@ abstract class Syncer<CollectionType: LocalCollection<*>>(
      *
      * @param localCollections The local collections to be updated or deleted
      * @param dbCollections The database collections possibly containing new information
-     * @return New found database collections to be created in provider
+     * @return Pair of:
+     *  1) New found database collections to be created in provider
+     *  2) Local collections which have been deleted
      */
     @VisibleForTesting
     internal fun updateCollections(
         localCollections: List<CollectionType>,
         dbCollections: Map<HttpUrl, Collection>
-    ): HashMap<HttpUrl, Collection> {
+    ): Pair<HashMap<HttpUrl, Collection>, List<CollectionType>> {
         val newDbCollections = HashMap(dbCollections)   // create a copy
+        val deletedLocalCollections = mutableListOf<CollectionType>()
         for (localCollection in localCollections) {
             val dbCollection = dbCollections[localCollection.collectionUrl?.toHttpUrlOrNull()]
-            if (dbCollection == null)
+            if (dbCollection == null) {
                 // Collection not available in db = on server (anymore), delete obsolete local collection
                 localCollection.deleteCollection()
```

Comment:
In order to have the actual deletion and the deletion from to-be-synced-list at the same place, this could happen outside of `updateCollections` where we delete from the list too.